plugins/de/opendiabetes/vault/plugin/importer/ODV/ODVImporter.java

```diff
@@ -0,0 +1,368 @@
+                return false;
+            }
+            Iterator iterator = metaInfo.entrySet().iterator();
+            PluginManager manager = new DefaultPluginManager();
+            manager.loadPlugins();
+            manager.startPlugins();
+            while (iterator.hasNext()) {
+                Map.Entry metaEntry = (Map.Entry) iterator.next();
+                String plugin = (String) metaEntry.getKey();
+                MetaValues metaValues = (MetaValues) metaEntry.getValue();
+                String importFile = tempDir + File.separator + metaValues.file;
+                Importer importer;
+                try {
+                    importer = (Importer) manager.getExtensions(plugin).get(0);
+                } catch (Exception exception) {
+                    LOG.log(Level.WARNING, "No Plugin named {0} available ", plugin);
+                    unimportedFiles.put(importFile, reasonNoPlugin);
+                    continue;
+                }
+                if (!verifyChecksum(importFile, metaValues.checksum)) {
+                    unimportedFiles.put(importFile, reasonChecksumFailed);
+                    continue;
+                }
+                importer.setImportFilePath(importFile);
+                importer.importData();
+                importedData.addAll(importer.getImportedData());
+                result = true;
+            }
+            notifyStatus(PROGRESS_AVAILABLE_IMPORTED, "Imported all available data");
+            reportUnimported(unimportedFiles);
+            return result;
+        }
+
+        /**
+         * This method loads the configuration and sets the values accordingly.
+         *
+         * @param configuration The configuration object that hold all configurations.
+         * @return True if the configuration could be loaded, false otherwise.
+         */
+        @Override
+        public boolean loadConfiguration(final Properties configuration) {
+            if (configuration.containsKey("temporaryDirectory")) {
+                this.tempDir = configuration.getProperty("temporaryDirectory");
+            }
+            if (configuration.containsKey("metaFile")) {
+                this.metaFile = configuration.getProperty("metaFile");
+            }
+            this.notifyStatus(PROGRESS_CONFIG_LOADAD, "Loaded configuration");
+            return true;
+        }
+
+        /**
+         * This method is used to check the integrity of the specified file by comparing it to the supplied checksum.
+         *
+         * @param filePath Path to the file to be checked.
+         * @param checksum The checksum to check the file against.
+         * @return True if the integrity could be verified successfully, false otherwise.
+         */
+        private boolean verifyChecksum(final String filePath, final String checksum) {
+            FileInputStream fileInputStream;
+            try {
+                fileInputStream = new FileInputStream(filePath);
```

Comment:
FindBugs bemängelt hier, dass der Stream nicht in allen Situationen geschlossen wird.

Reply:
Ja, aber das Problem hierbei ist, dass wenn die FileNotFound geworfen wird ist fileInputStream (fis) logischerweise null, von daher geht kein close()
Habs über if(fis != null) -> close() versucht, dann sagt mir Intellij logischerweise, dass fis immer null ist. Sieht mir nach nem False Positive von FB aus, die Regel ist ja auch als Experimental gemarked

Reply:
Alright, kein Problem. Hatte das auch bei dem normalen InputStream dort stand aber im JavaDoc dass bei InputStreams close() nichts macht

Reply:
Ok, dann würde ich das lassen und als False Positive markieren

Reply:
👍